client/my-sites/site-settings/jetpack-credentials/index.jsx

```diff
@@ -0,0 +1,313 @@
+			case 'PRESSABLE-SSH':
+				return translate( 'A special Secure Shell connection to Pressable.' );
+		}
+
+		return '';
+	};
+
+	autoConfigure = () => this.props.autoConfigCredentials( this.props.siteId );
+
+	renderFormFoldable() {
+		const {
+			mainCredentials,
+			translate,
+			isPressable
+		} = this.props;
+
+		const protocol = get( mainCredentials, 'protocol', 'SSH' ).toUpperCase();
+		const description = this.getProtocolDescription( protocol );
+
+		const header = (
+			<div className="jetpack-credentials__header">
+				<Gridicon icon="checkmark-circle" size={ 48 } className="jetpack-credentials__header-gridicon" />
+				<div className="jetpack-credentials__header-text">
+					<h3 className="jetpack-credentials__header-protocol">{ protocol }</h3>
+					<h4 className="jetpack-credentials__header-description">{ description }</h4>
+				</div>
+			</div>
+		);
+
+		if ( isPressable ) {
+			return (
+				<CompactCard className="jetpack-credentials__pressable-configured">
+					<Gridicon icon="checkmark-circle" size={ 48 } className="jetpack-credentials__header-gridicon" />
+					<div className="jetpack-credentials__header-configured-text">
+						{ translate( 'You\'re all set! Your credentials have been ' +
+							'automatically configured and your site is connected. ' +
+							'Backups and restores should run seamlessly.'
+						) }
+					</div>
+				</CompactCard>
+			);
+		}
+
+		return (
+			<FoldableCard
+				header={ header }
+				className="jetpack-credentials__foldable-header"
+			>
+				{ this.renderForm() }
+			</FoldableCard>
+		);
+	}
+
+	renderForm() {
+		const {
+			credentialsUpdating, // eslint-disable-line no-shadow
+			siteId,
+			updateCredentials,
+		} = this.props;
+
+		return (
+			<CredentialsForm { ...{
+				credentialsUpdating,
+				protocol: get( this.props.mainCredentials, 'protocol', 'ssh' ),
+				host: get( this.props.mainCredentials, 'host', '' ),
+				port: get( this.props.mainCredentials, 'port', '' ),
+				user: get( this.props.mainCredentials, 'user', '' ),
+				pass: get( this.props.mainCredentials, 'pass', '' ),
+				abspath: get( this.props.mainCredentials, 'abspath', '' ),
+				kpub: get( this.props.mainCredentials, 'kpub', '' ),
+				onCancel: this.resetSetup,
+				siteId,
+				updateCredentials,
+			} } />
+		);
+	}
+
+	renderSetupStart() {
+		const {
+			translate
+		} = this.props;
+
+		return (
+			<CompactCard
+				className="jetpack-credentials__setup-start"
+				onClick={ this.goToNextSetupStep }
+			>
+				<Gridicon icon="add-outline" size={ 48 } className="jetpack-credentials__header-gridicon" />
+				<div className="jetpack-credentials__header-text">
+					<h3 className="jetpack-credentials__header-text-title">{ translate( 'Add site credentials' ) }</h3>
+					<h4 className="jetpack-credentials__header-text-description">
+						{ translate( 'Used to perform automatic actions on your server including backing up and restoring.' ) }
+					</h4>
+				</div>
+			</CompactCard>
+		);
```

Comment:
this definitely looks like a good candidate to extract into a separate component